client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -743,13 +749,14 @@ class CalypsoifyIframe extends Component<
 					source=""
 					visible={ isMediaModalVisible }
 				/>
-				{ isCheckoutOverlayEnabled && isCheckoutModalVisible && (
+				{ isCheckoutOverlayEnabled && (
```

Comment:
allow pre-loading Checkout Modal before opening it

Reply:
What are the perf consequences of this? We may be loading the checkout modal even if we don't need it

Reply:
There are considerably more cases when we need it than when we don't. For example:
- All free and personal sites where a premium block has been added.
- All unlaunched sites (even those with paid plans could use checkout to add a domain).

Of course we can be more specific when to load it but until then, if initial load time isn't affected, the overall improvement in UX overweights any downside from my perspective.

Cc'ing @tyxla for confirmation on perf impact.

Reply:
From a perf standpoint I see no concerns here. What has been necessary here has already been done - the checkout modal is loaded asynchronously and that's enough to defer the loading for after the initial page load. In other words, initial loading time won't be affected by this because we're already asynchronously loading the checkout modal. 

> Of course we can be more specific when to load it but until we add all the conditions above, if initial load time isn't affected, the overall improvement in UX overweights any downside from my perspective.

Ditto. I agree the better user experience of seeing the checkout modal immediately is more important than the few extra bytes they might end up downloading in background, as long as the initial load speed is unaffected.

Reply:
Thank you both for the explanation. We can consider this review comment resolved.